app/components/contacts/AllContacts.js

```diff
@@ -57,8 +58,10 @@ class AllContacts extends Component<Props, State> {
   };
 
   render() {
-    const { contacts, lastUsedAddresses, addContact, selectContact, isModalMode } = this.props;
+    const { contacts, transactions, addContact, selectContact, isModalMode } = this.props;
     const { searchTerm } = this.state;
+    const lastUsedAddresses: TxList = uniqBy(transactions, 'address').slice(0, 3);
```

Comment:
think of transaction list containing 5000 entries. uniqBy will run through the whole list making comparison just to get 3 last used unique addresses. And this will happen every render call. Is this best solution?

Reply:
changed to catch the first 3 unique and break when that happens.